### ISSUE TITLE
chore(build): script for publishing to GH pages

### ISSUE THREAD

**Issue by btford** [angular/router#55]
Right now, I'm manually doing something like this:

```shell
git reset --hard HEAD
git checkout gh-pages
gulp build
cp -R dist/docs/* ./
git add -A *.html api/*.html content/*.html
git commit -m 'update docs'
git push upstream gh-pages
```

**Comment by jvandemo** [comment]
@btford &mdash; I wrote a once to deploy Harp websites to a gh-pages branch using Travis CI: https://github.com/jvandemo/deploy-harp-to-gh-pages-using-travis/blob/master/scripts/deploy.sh

If you are also interested in using Travis CI for automatically generating GH pages, I am willing to invest some time in this to help you out.

Feel free to let me know if you are interested to get a discussion going on the details.

**Comment by btford** [comment]
@jvandemo – I like the idea, except I'd like to only push if:

1. it's tagged as a release
2. tests pass

**Comment by jvandemo** [comment]
@btford &mdash; Ok, I'll dig into it.

A few questions to make sure I approach things correctly:

1. **tagged as release**: I assume that this means all tags that look like v[0-9].[0-9].[0-9]? (We can match the tag against a regular expression so Travis CI ignores builds that do not match the tag.)
2. **when tests have passed**: no problem here, we can run `npm test` and deploy new documentation if it succeeds. That sounds good?
3. **repository**: do you want to deploy the documentation to the `gh-pages` branch of `angular/router` or another repository?

Feel free to add items if necessary.

Thanks!

**Comment by btford** [comment]
1. `v[0-9]+.[0-9]+.[0-9]+`, but otherwise correct
2. Sounds good!
3. I want to deploy to the `gh-pages` branch of this repo (`angular/router`)

Thanks!`

**Comment by jvandemo** [comment]
@btford &mdash; Quick update: things are starting to work, however the `gulp dgeni` seems to throw an error when run on Travis CI:

![schermafbeelding 2015-03-13 om 18 32 27](https://cloud.githubusercontent.com/assets/1859381/6643806/8aff0e30-c9af-11e4-96a5-a4d47251516e.png)

I'm not experiencing this locally, so will try and figure out what is causing this.

I'm also preparing documentation on how an GitHub token can be configured to perform the deployment:
https://github.com/jvandemo/router/blob/feat/automate-docs-with-travis/DOCUMENTATION.md

Will keep you posted on the progress.

**Comment by jvandemo** [comment]
@petebacondarwin &mdash; Have you experienced problems with running dgeni on Travis CI before?

I am trying to figure out what is happening and would really appreciate your thoughts on this.

Thanks in advance!

**Comment by jvandemo** [comment]
Added the angular/router dgeni configuration to a brand new repository that runs dgeni on Travis CI, and get exactly the same error:

https://travis-ci.org/jvandemo/run-dgeni-on-travis-ci/builds/54364008

Running fine on my MacBook, but not on Travis CI.

The `.travis.yml` file in this new repository only contains dgeni tasks so I can troubleshoot quicker and builds run faster on Travis.

I'll keep digging into it...

**Comment by jvandemo** [comment]
Commenting out:

```javascript
package.processor(require('./processors/markdown.js'));
```

in `dgeni.conf`seems to prevent the problem but also prevents output being generated.

Digging further...

**Comment by jvandemo** [comment]
Quick update: figured it out and got it to work. Will document and update the code tomorrow.

**Comment by jvandemo** [comment]
@btford &mdash; I have created PR #180 with code that should hopefully solve this issue.

The only thing left is to update the `GITHUB_TOKEN` variable in `.travis.yml` with a token that has access to the `angular/router` repository since my token doesn't have access.

Documentation on how to add the `GITHUB_TOKEN` variable has been added here: https://github.com/jvandemo/router/blob/feat/automate-docs-with-travis/DOCUMENTATION.md.

If you need any help or something goes wrong, please don't hesitate to contact me.

**Comment by btford** [comment]
Landed with #180
